rs/nervous_system/clients/src/request.rs

```diff
@@ -0,0 +1,10 @@
+use candid::CandidType;
+use serde::de::DeserializeOwned;
+
+pub trait Request: CandidType {
```

Comment:
I'm not quite seeing that it is useful outside of the agent crate. Do you see it being used in canister calls and state machine tests?

Reply:
That is exactly what I had in mind :smile: we actually already have something very similar in the ic repo for the system tests, but with a bit more boilerplate than this design. But my long term goal is to upgrade our canister code to use `Request` impls rather than encoding/decoding every time, and same for tests